### DISCUSSION TITLE
ETCD failing while okd cluster install - bare metal (disconnected) ERROR: https://localhost:2379 is unhealthy

### DISCUSSION THREAD

**Discussion by mohitagarwal1** [okd-project/okd#1975] [Q&A]
Hi everyone,
I am trying to build OKD cluster on bare metal. The bootstrap is stuck on the below error:
https://localhost:2379 is unhealthy: failed to commit proposal: context deadline exceeded
Error: unhealthy cluster

I am using the following fedora coreos image:

https://builds.coreos.fedoraproject.org/prod/streams/stable/builds/40.20240701.3.0/x86_64/fedora-coreos-40.20240701.3.0-live-initramfs.x86_64.img

https://builds.coreos.fedoraproject.org/prod/streams/stable/builds/40.20240701.3.0/x86_64/fedora-coreos-40.20240701.3.0-live-rootfs.x86_64.img

https://builds.coreos.fedoraproject.org/prod/streams/stable/builds/40.20240701.3.0/x86_64/fedora-coreos-40.20240701.3.0-live-kernel-x86_64

os version : 40.20240701.3.0
release version : 4.14.0-0.okd-2024-01-26-175629

I have found same issue on bugzilla redhat : https://bugzilla.redhat.com/show_bug.cgi?id=1845970
But no proper resolutions there.

Please help to provide your input on how I can overcome this error.

**Comment by bdlink** [comment]
After downloading the correct openshift-install, run
openshift-install coreos print-stream-json
to find the initial boot image to use.  For the version of okd you are trying to install, it should be fedora-coreos-38.20231012.3.1

Using a different major version of fcos is likely to be problematic.  The voluminous output also gives you a url to pull the right version of fcos.

Please try this and see if it helps.

If you are trying the fcos branch, there is no reason not to use the latest 4.15 stable.  It uses fedora-coreos-39.20231101.3.0

On Jul 17, 2024, at 10:53 AM, Mohit Agarwal ***@***.***> wrote:

﻿
CAUTION: This email originated from outside of BCIT. Do not click links or open attachments unless you recognize the sender and know the content is safe.


Hi everyone,
I am trying to build OKD cluster on bare metal. The bootstrap is stuck on the below error:
https://localhost:2379 is unhealthy: failed to commit proposal: context deadline exceeded
Error: unhealthy cluster

I am using the following fedora coreos image:

https://builds.coreos.fedoraproject.org/prod/streams/stable/builds/40.20240701.3.0/x86_64/fedora-coreos-40.20240701.3.0-live-initramfs.x86_64.img

https://builds.coreos.fedoraproject.org/prod/streams/stable/builds/40.20240701.3.0/x86_64/fedora-coreos-40.20240701.3.0-live-rootfs.x86_64.img

https://builds.coreos.fedoraproject.org/prod/streams/stable/builds/40.20240701.3.0/x86_64/fedora-coreos-40.20240701.3.0-live-kernel-x86_64

os version : 40.20240701.3.0
release version : 4.14.0-0.okd-2024-01-26-175629

I have found same issue on bugzilla redhat : https://bugzilla.redhat.com/show_bug.cgi?id=1845970
But no proper resolutions there.

Please help to provide your input on how I can overcome this error.

**Reply by mohitagarwal1** [reply]
Thanks for the response. I have tried with the version you mentioned but still no luck. Meanwhile I am testing, could you please help with the workable version which I can use for now to test whether the installation works in my setup environment.

Please help to provide workable version for okd, and coreos.

Thanks

**Reply by mohitagarwal1** [reply]
Hi @bdlink 
I tried openshift-install coreos print-stream-json to find the initial boot image to use, and then changed the image accordingly.
But still I am facing with the same issue with ETCD. 

I used the following release and coreos:
release: 38.20231002.3.1
okd version 4.14.0-0.okd-2024-01-26-175629 

Please help to provide details on how can I start with it, or any reference? I need to setup a okd cluster in my environment. It should be a disconnected bare metal installation. Please help on this.
